detectron2/engine/hooks.py

```diff
@@ -199,6 +201,81 @@ def after_step(self):
         self.step(self.trainer.iter)
 
 
+class BestCheckpointer(HookBase):
+    """
+    Checkpoints best weights based off given metric.
+
+    This hook should be used in conjunction to and executed after `EvalHook`.
+    """
+
+    def __init__(
```

Comment:
please add an option `mode="max"` similar to the one in https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/ModelCheckpoint, with two possible values "min" and "max"

Reply:
oh right, sorry! totally skipped my mind that the metric could go the other direction :facepalm: